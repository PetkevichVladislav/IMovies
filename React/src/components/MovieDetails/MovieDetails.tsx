import "./MovieDetails.scss"

import Logo from "../Logo/Logo";
import { MovieModel } from "../../models/MovieModel";
import { Link, useSearchParams } from "react-router-dom";
import { FC, useMemo } from "react";

export interface IMovieDetails {
    movie?: MovieModel | null;
}

export const MovieDetails: FC<IMovieDetails> = ({ movie: details }) => {
    const [searchParams] = useSearchParams();
    const link = useMemo(() => "/?" + searchParams.toString(), [searchParams]);
    
    return (
        <div className="movie-details">
            <div className="movie-details__icons-container">
                <div className="movie-details__logo-container">
                    <Logo />
                </div>
                <Link to={link}>
                    <span className="icon-search" />
                </Link>
            </div>
            <div className="movie-details__info-container">
                <figure className="movie-details__image-container">
                    <img className="movie-details__image" src={details?.imagePath} alt="movie_card"></img>
                </figure>
                <div className="movie-details__details-container">
                    <div className="movie-details__heading-container">
                        <p className="movie-details__details-information--title">{details?.title}</p>
                        {details?.rating && <p className="movie-details__details-information--score">{details?.rating}</p>}
                    </div>
                    <p className="movie-details__details-information--genre">{details?.genres.join(", ")}</p>
                    <p className="movie-details__details-information--release-date">{details?.releaseDate.getFullYear()}</p>
                    {details?.runtime && <p className="movie-details__details-information--duration">{details?.runtime} minutes</p>}
                    <p className="movie-details__details-information--description">{details?.overview}</p>
                </div>
            </div>
        </div>
    );
};