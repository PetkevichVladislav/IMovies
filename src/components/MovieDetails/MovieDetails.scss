.movie-details {
    background-color: #232323;
    max-height: 600px;
    padding: 40px 60px 30px 60px;

    &__icon {
        &--search {
            &::before {
                content: "\e900";
                color: #F65261;
                font-size: 15px;
            }
        }
    }

    &__icons-container {
        display: flex;
        justify-content: space-between;
    }

    &__info-container {
        display: flex;
        margin-top: 30px;
    }

    &__image-container {
        width: 323px;
        max-height: 486px;
    }

    &__image {
        background-size: cover;
        background-position: center;
        display: flex;
        justify-content: center;
        align-items: center;
        width: inherit;
    }

    &__heading-container {
        display: flex;
    }

    &__details-container {
        margin-left: 50px;
    }

    &__details-information {

        &--title {
            display: inline-flex;
            color: #FFFFFF;
            font-size: 40px;
            margin: 0px;
        }

        &--score {
            display: inline-flex;
            color: #FFFFFF;
            justify-content: center;
            width: 50px;
            height: 50px;
            /* min-width: fit-content; */
            align-items: center;
            border: 1px solid #FFFFFF;
            /* box-sizing: revert; */
            border-radius: 50%;
            /* padding: 34px; */
            font-size: 20px;
            align-self: center;
            margin: 0px 0px 0px 25px;
        }

        &--genre {
            color: #FFFFFF;
            opacity: 0.5;
            margin: 9px 0px 30px 0px;
        }

        &--release-date {
            display: inline-block;
            color: #F65261;
            margin: 0px;
        }

        &--duration {
            display: inline-block;
            color: #F65261;
            margin: 0px 0px 0px 50px;
        }

        &--description {
            color: #FFFFFF;
            opacity: 0.5;
            margin: 30px 0px 0px 0px;
        }
    }
}

.icon-search {
    transform: rotateY(180deg);
    cursor: pointer;

    &::before{
        content: "\e902";
        color: #F65261;
        font-size: 30px;
    }
}