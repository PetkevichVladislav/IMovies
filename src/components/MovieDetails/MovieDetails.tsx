import "./MovieDetails.scss"
import { FC } from "react"

import Logo from "../Logo/Logo";

const bottomBorderScore = 0;
const topBottomBorderScore = 10;

export interface IMovieDetails {
    imageUrl: string;
    movieName: string;
    movieGenres: string[];
    releaseYear: string;
    rating: number;
    duration: string;
    description: string;
}

export const MovieDetails: FC<IMovieDetails> = (details: IMovieDetails) => {
    let score = details.rating > topBottomBorderScore ? 10 : details.rating;
    score = score < bottomBorderScore ? bottomBorderScore : score

    return (
        <div className="movie-details">
            <div className="movie-details__icons-container">
                <div className="movie-details__logo-container">
                    <Logo />
                </div>
                <span className="icon-search"></span>
            </div>
            <div className="movie-details__info-container">
                <figure className="movie-details__image-container">
                    <img className="movie-details__image" src={details.imageUrl} alt="movie_card"></img>
                </figure>
                <div className="movie-details__details-container">
                    <div className="movie-details__heading-container">
                        <p className="movie-details__details-information--title">{details.movieName}</p>
                        <p className="movie-details__details-information--score">{score}</p>
                    </div>
                    <p className="movie-details__details-information--genre">{details.movieGenres.join(", ")}</p>
                    <p className="movie-details__details-information--release-date">{details.releaseYear}</p>
                    <p className="movie-details__details-information--duration">{details.duration}</p>
                    <p className="movie-details__details-information--description">{details.description}</p>
                </div>
            </div>
        </div>
    );
};