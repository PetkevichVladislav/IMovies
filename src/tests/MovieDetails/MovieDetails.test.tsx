import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { IMovieDetails, MovieDetails } from '../../components/MovieDetails/MovieDetails';

describe("Movie details", () => {
    const movieData : IMovieDetails = {
        imageUrl: "http://uimage.com",
        movieName: 'Example Movie',
        releaseYear: '2022',
        movieGenres: ['Action', 'Drama'],
        description: "Movie description",
        duration: "2h 35min",
        rating: 9.3,
    };

    it("When data is valid component should renders correctly", () =>{
        const tree = renderer.create(<MovieDetails {...movieData}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();;
    });

    it("when user open forms the data should reneders data", () =>{
        render(<MovieDetails {...movieData}/>);
        expect(screen.getByText('Example Movie')).toBeInTheDocument();
        expect(screen.getByText('2022')).toBeInTheDocument();
        expect(screen.getByText('Action, Drama')).toBeInTheDocument();
        expect(screen.getByText("Movie description")).toBeInTheDocument();
        expect(screen.getByText("2h 35min")).toBeInTheDocument();
        expect(screen.getByText("9.3")).toBeInTheDocument();
    });
});