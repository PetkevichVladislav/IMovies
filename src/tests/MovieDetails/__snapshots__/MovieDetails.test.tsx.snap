// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie details When data is valid component should renders correctly 1`] = `
<div
  className="movie-details"
>
  <div
    className="movie-details__icons-container"
  >
    <div
      className="movie-details__logo-container"
    >
      <div
        className="logo"
      >
        <p
          className="logo__text logo__text--bold"
        >
          i
        </p>
        <p
          className="logo__text logo__text--thin"
        >
          movie
        </p>
      </div>
    </div>
    <span
      className="icon-search"
      onClick={[Function]}
    />
  </div>
  <div
    className="movie-details__info-container"
  >
    <figure
      className="movie-details__image-container"
    >
      <img
        alt="movie_card"
        className="movie-details__image"
      />
    </figure>
    <div
      className="movie-details__details-container"
    >
      <div
        className="movie-details__heading-container"
      >
        <p
          className="movie-details__details-information--title"
        />
        <p
          className="movie-details__details-information--score"
        />
      </div>
      <p
        className="movie-details__details-information--genre"
      />
      <p
        className="movie-details__details-information--release-date"
      />
      <p
        className="movie-details__details-information--duration"
      >
         minutes
      </p>
      <p
        className="movie-details__details-information--description"
      />
    </div>
  </div>
</div>
`;
